src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -531,18 +532,18 @@ static int server_send_ack(
                         return r;
         }
 
-        ORDERED_HASHMAP_FOREACH(j, server->extra_options) {
+        ORDERED_SET_FOREACH(j, server->extra_options) {
                 r = dhcp_option_append(&packet->dhcp, req->max_optlen, &offset, 0,
                                        j->option, j->length, j->data);
                 if (r < 0)
                         return r;
         }
 
-        if (!ordered_hashmap_isempty(server->vendor_options)) {
+        if (!ordered_set_isempty(server->vendor_options)) {
                 r = dhcp_option_append(
                                 &packet->dhcp, req->max_optlen, &offset, 0,
                                 SD_DHCP_OPTION_VENDOR_SPECIFIC,
-                                ordered_hashmap_size(server->vendor_options), server->vendor_options);
+                                ordered_set_size(server->vendor_options), server->vendor_options);
```

Comment:
Please also update the case `SD_DHCP_OPTION_VENDOR_SPECIFIC` in dhcp_option_append().